pkg/networkservice/common/mechanisms/memif/client.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package memif provides the necessary mechanisms to request and inject a memif socket.
+package memif
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/memif"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+type memifMechanismClient struct {
+	socketFileName string
+}
+
+// NewClient - returns client that sets memif preferred mechanism
+func NewClient(socketFileName string) networkservice.NetworkServiceClient {
```

Comment:
Generally speaking, clients are *always* consumers of socket files in memif, *never* providers.  Do we have a use case in which we need it to be otherwise?